src/ray/core_worker/task_event_buffer.cc

```diff
@@ -143,72 +133,67 @@ void TaskEventBufferImpl::FlushEvents(bool forced) {
       return;
     }
 
-    if (buffer_.size() == 0) {
+    // No data to send.
+    if (buffer_.empty()) {
       return;
     }
 
-    task_events.reserve(
-        RayConfig::instance().task_events_max_num_task_events_in_buffer());
-    buffer_.swap(task_events);
-    next_idx_to_overwrite_ = 0;
+    size_t num_to_send =
+        std::min(RayConfig::instance().task_events_send_batch_size(), buffer_.size());
+    to_send.insert(to_send.end(), buffer_.begin(), buffer_.begin() + num_to_send);
```

Comment:
Doesn't this unnecessarily copy the data? Is it feasible to directly swap each value in the buffer to td::make_unique<rpc::TaskEventData>();?

E.g., 

```
for (auto it : buffer_) {
    data->Swap(it)
}
```

or 

```
auto val = std::move(buffer_.pop_front());
data->Swap(val)
```
(not sure if it is visible)